src/backend/embedding/extract_file/extract_file.py

```diff
@@ -0,0 +1,173 @@
+            extension = MIME_TO_EXTENSION.get(file_type)
+            if not extension:
+                logger.error(f"Unsupported file type detected: {file_type}")
+                raise HTTPException(status_code=415, detail="Unsupported file type.")
+            return extension
+        except Exception as e:
+            logger.error(f"Error detecting file type for {file_path}: {e}")
+            raise HTTPException(status_code=400, detail="File type detection failed.")
+
+    def format_table_as_text(self, table):
+        try:
+            if table is None or not table:
+                logger.error("Empty or None table received for formatting.")
+                return "Empty table"
+
+            table = [[str(item) if item is not None else "" for item in row] for row in table]
+            max_columns = max(len(row) for row in table)
+            table = [row + [""] * (max_columns - len(row)) for row in table]  # Fill missing cells with empty strings
+            col_widths = [max(len(str(item)) for item in col) for col in zip(*table)]
+            row_format = " | ".join(f"{{:<{w}}}" for w in col_widths)
+            formatted_table = "\n".join(row_format.format(*row) for row in table)
+            return formatted_table.strip()
+
+        except Exception as e:
+            logger.error(f"Error formatting table: {e}")
+            return "Error occurred while formatting table."
+
+    def extract_pdf_content(self, customer_guid: str, file_path: str, filename: str):
+        results = []
+        try:
+            with pdfplumber.open(file_path) as pdf:
+                for page_number, page in enumerate(pdf.pages, start=1):
+                    page_elements = []
+
+                    for block in page.extract_words():
+                        page_elements.append({
+                            "type": "text",
+                            "content": block["text"],
+                            "x0": block["x0"],
+                            "y0": block["top"]
+                        })
+
+                    for table in page.extract_tables():
+                        table_text = self.format_table_as_text(table)
+                        bbox = page.find_tables()[0].bbox  # Get bounding box for the first table
+                        page_elements.append({
+                            "type": "table",
+                            "content": table_text,
+                            "x0": bbox[0],
+                            "y0": bbox[1]
+                        })
+
+                        pil_image = page.to_image(resolution=300).original
+                        numpy_image = np.array(pil_image)
+                        ocr_result = self.ocr_reader.readtext(numpy_image)
+                        ocr_text = " ".join([text[1] for text in ocr_result])
+                        page_elements.append({
+                        "type": "image",
+                        "content": f"Data obtained from image: {ocr_text.strip()}",
```

Comment:
Lets not have the words "Data obtained from image"
Just add the content like "content": ocr_text.strip()